checker/tests/nullness/PolyAllTest.java

```diff
@@ -1,12 +1,18 @@
 import org.checkerframework.checker.nullness.qual.NonNull;
 import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.nullness.qual.PolyNull;
 import org.checkerframework.dataflow.qual.Pure;
 import org.checkerframework.framework.qual.PolyAll;
 
 public class PolyAllTest<T extends Comparable<T>> {
 
     @Pure
-    public int compare(@PolyAll T[] a1, @PolyAll T[] a2) {
+    public int comparePolyAll(@PolyAll T[] a1, @PolyAll T[] a2) {
```

Comment:
If `@PolyAll` is written on a use of a type variable, then it is ignored.  Ordinarily, a type system can specify a different annotation than `@PolyAll` by with an explicit qualifier, but in the case of type variables, there is no way to specify a different qualifier.

Reply:
I don't understand this statement and find the difference in error messages between `comparePolyAll` and `compare` very confusing.
I would expect that both methods behave the same, but they don't.
Why are `PolyNull` and `PolyAll` different on type variables?
Is this described in the manual?

What do you mean with `there is no way to specify a different qualifier`? What qualifier would you want to specify?

Reply:
I've made the behavior consistent between `@PolyAll` and `@PolyNull`.  (Note, this pull request is not ready for review yet.)